bloom-housing/bloom
# fix: allow for more space for modals on mobile

# Pull Request Template

## Issue Overview

This PR addresses #3062


## Description

This PR makes some UI fixes and changes in the Modal component.

* Update spacing according to the Figma to be a consistent 24px
* On mobile, remove the border radius from modals
* On mobile, allow modals to take up the full height and width of the screen if space is needed + scroll the full modal in these cases 
* In public/partners, default to using the small button style in modals and drawers
* Pulls over a fix from Detroit that was preventing modals from scrolling on mobile iOS
* If buttons overflow in the footer, wrap them instead of cutting them off
* Cancel buttons are now the default styling with the white background versus borderless (screenshots below)

<img width="518" alt="Screen Shot 2022-11-01 at 4 16 22 PM" src="https://user-images.githubusercontent.com/65367387/199332568-229acbd9-1f80-4cb6-a9d6-1d366652c6a5.png">
<img width="520" alt="Screen Shot 2022-11-01 at 4 16 32 PM" src="https://user-images.githubusercontent.com/65367387/199332573-5b5e288c-5cdc-43ba-834c-a7f33f4e5e2b.png">

## How Can This Be Tested/Reviewed?

Review all of the [modal stories](https://deploy-preview-3153--dev-storybook-bloom.netlify.app/?path=/story/overlays-modal--basic-modal) at all four of the Storybook viewports (default, tablet, large mobile, small mobile).

Ensure that when content is overflowing on mobile, the modal takes up the full width of the page and is scrollable.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
